app/components/paperShow/index.tsx

```diff
@@ -815,11 +753,64 @@ class PaperShow extends React.PureComponent<PaperShowProps, PaperShowStates> {
 
   private fetchComments = (page: number = 1) => {
     const { paper, dispatch } = this.props;
-
     if (paper) {
       dispatch(getComments({ paperId: paper.id, page }));
     }
   };
+
+  private fetchPapersInCollection = (collectionId: number) => {
+    const { myCollections, dispatch } = this.props;
+    if (myCollections.length > 0) {
+      dispatch(getPapers(collectionId));
+    }
+  };
+
+  private getMyCollections = async () => {
+    const { dispatch, currentUser, paper } = this.props;
+    if (currentUser && currentUser.isLoggedIn && (currentUser.oauthLoggedIn || currentUser.emailVerified)) {
```

Comment:
You can use `currentUser.isLoggedIn && currentUser.emailVerified`.
And I think if a user isn't logged in status, you can use `app/helpers/checkAuthDialog.ts` method.

I'll make the logic that handles not-verified user later.